### ISSUE TITLE
IntegerUidProviders are not singletons in hypnotoad with multiple workers

### ISSUE THREAD

**Issue by vikin91** [vikin91/BibSpace#10]
```
[Tue Feb  7 10:20:30 2017] WARNING: Id Provider for Tag          has addr '140607898259360', last_id: 231 (Origin: SmartUidProvider->get_provider).
...
[Tue Feb  7 10:20:33 2017] WARNING: Id Provider for Tag          has addr '140607891694720', last_id: 0 (Origin: SmartUidProvider->get_provider).
[Tue Feb  7 10:20:33 2017] DEBUG: IntegerUidProvider (140607891694720) has generated uid '1' for type 'Tag' (Origin: IntegerUidProvider->generateUID)
```
The address in memory of Tag ID Provider changes during operation...

**Comment by vikin91** [comment]
Possibly-useful hints: https://groups.google.com/forum/#!msg/mojolicious/T9uH9a14CHM/Ujqpc2EPBgAJ

**Comment by vikin91** [comment]
Starting BibSpace like this causes no problems:
```
BIBSPACE_USE_DUMP=1  bin/bibspace daemon -m production -l http://*:8080
BIBSPACE_USE_DUMP=1  bin/bibspace daemon -m production -w 2 -l http://*:8080
BIBSPACE_USE_DUMP=1  bin/bibspace prefork -m production -w 2 -l http://*:8080
BIBSPACE_USE_DUMP=1  bin/bibspace prefork -m production -w 4 -c 10  -l http://*:8080
```
but using hypnotoad do cause problems - even with single worker.

**Comment by vikin91** [comment]
`bin/bibspace daemon -l http://*:8080` causes problems on linux as well... this must be something else.

**Comment by vikin91** [comment]
Next trait: this problem seems to be related with `USE_BISPACE_DUMP` and restoring backups. By restoration, the UIdProvider seems to be duplicated.

**Comment by vikin91** [comment]
The problem seems to be located in:
```
my $drh = DBI->install_driver("mysql");
my $rc = $drh->func( 'dropdb', $db_database, $db_host, $db_user, $db_pass, 'admin' );
my $rc = $drh->func( 'createdb', $db_database, $db_host, $db_user, $db_pass, 'admin' );
```
Need more tests to make sure it is fixed now.

**Comment by vikin91** [comment]
Starting the app on the test server using command:
`BIBSPACE_CONFIG=/etc/bibspace_test.conf BIBSPACE_USE_DUMP=0 MOJO_MODE=development hypnotoad -f bin/bibspace` seems to cause no problems. 
Starting it with `sudo service bibspace-test start` (runs as root) causes problems. 
Maybe the issue is caused by sudo/root?

**Comment by vikin91** [comment]
It seemed to be a problem with keeping some objects inside mojolicious `has` function. The function is executed only once at startup and during layer replacement the original object is not replaced with the new one, but exists alongside with it. This seems to be fixed finally in 4a0a63e

**Comment by vikin91** [comment]
http://search.cpan.org/~sri/Mojolicious-7.25/lib/Mojolicious/Guides/Cookbook.pod#DEPLOYMENT
